src/rate_limiter.rs

```diff
@@ -21,28 +21,35 @@ use error::RoutingError;
 use fake_clock::FakeClock as Instant;
 use itertools::Itertools;
 use maidsafe_utilities::serialisation::{self, SerialisationError};
-use messages::UserMessage;
+use messages::{MAX_PART_LEN, UserMessage};
 use sha3::Digest256;
 use std::cmp;
 use std::collections::BTreeMap;
+use std::collections::btree_map::Entry;
 use std::mem;
 use std::net::IpAddr;
 #[cfg(not(feature = "use-mock-crust"))]
 use std::time::Instant;
+use types::MessageId;
 
 /// The number of bytes per second the `RateLimiter` will "leak".
 const RATE: f64 = 5.0 * 1024.0 * 1024.0;
-/// The maximum number of bytes a single client is allowed to have in the `RateLimiter`.  This is
-/// slightly larger than `MAX_IMMUTABLE_DATA_SIZE_IN_BYTES` to allow for the extra bytes created by
-/// wrapping the chunk in a `UserMessage`, splitting it into parts and wrapping those in
-/// `RoutingMessage`s.
-const CLIENT_CAPACITY: u64 = MAX_IMMUTABLE_DATA_SIZE_IN_BYTES + 10240;
+/// The number of bytes per second of bandwidth distributed between all clients when demand is low.
+#[cfg(not(any(test, feature = "use-mock-crust")))]
```

Comment:
will `#[cfg(not(feature = "use-mock-crust"))]` and `#[cfg(feature = "use-mock-crust")]` do the same?